core/src/main/scala/org/apache/spark/scheduler/ExecutorLossReason.scala

```diff
@@ -37,6 +37,8 @@ private[spark] object ExecutorExited {
   }
 }
 
+private [spark] object LossReasonPending extends ExecutorLossReason("Pending loss reason.")
```

Comment:
As I explained in previous comments, I don't like this approach to the problem (the previous version doesn't use this class, instead there's an explicit `disableExecutor()` method in the scheduler). I'll wait for Kay to take a look before applying any feedback, in case she's ok with me just reverting my last commit.